### PR TITLE
Fix for table component checkbox selection focus the previous active …

### PR DESCRIPTION
**Pull request by ssreerama**
…row instead of newly checked row

This PR fixes #
1. Table component row focus issue on checkbox selection. Issue: when a table has multiple rows with checkbox component and any checkbox is selected, the table is to focus the previously selected row, but it should select the recently selected row.
2. This change fixes the problem with deployment options tab/ object type options tab in SC and SQL DB extensions, and SC diff window row selections.
fixes https://github.com/microsoft/azuredatastudio/issues/20202
fixes https://github.com/microsoft/azuredatastudio/issues/16718
fixes https://github.com/microsoft/azuredatastudio/issues/19674
fixes https://github.com/microsoft/azuredatastudio/issues/20280

https://user-images.githubusercontent.com/74571829/184026438-99d96879-5e1b-4d62-9b04-5b8e05c4755c.mp4

### CODE REVIEW

**Review by SakshiS-harma — APPROVED** [review]

**Comment by kisantia** [comment]
is #20280 also fixed by this?

File: src/sql/base/browser/ui/table/plugins/checkboxSelectColumn.plugin.ts [hunk 1]
```diff
@@ -191,7 +191,8 @@ export class CheckboxSelectColumn<T extends Slick.SlickData> implements Slick.Pl
 			this._grid.render();
 		}
 
-		//Ensure that the focus stays correct
+		//Ensure that the focus stays on current selected checkbox row
```
Comment (kisantia) [inline src/sql/base/browser/ui/table/plugins/checkboxSelectColumn.plugin.ts:194]: nit: isn't this making sure the focus is specifically on the checkbox's cell, not row?

**Review by kisantia — APPROVED** [review]

File: src/sql/base/browser/ui/table/plugins/checkboxSelectColumn.plugin.ts (hunk 1 above)
Comment (ssreerama) [reply src/sql/base/browser/ui/table/plugins/checkboxSelectColumn.plugin.ts]: yeah it focus the cell! updated

**Comment by ssreerama** [comment]
> is #20280 also fixed by this? Also, use "fixes" before the each of the issues mentioned in the description so that they get closed when this PR is merged :)

This what we created yesterday, forgot to mention this one, thanks!

**Review by aasimkhan30 — APPROVED** [review]

**Review by alanrenmsft — APPROVED** [review]

**Comment by ssreerama** [comment]
@chlafreniere /@alanrenmsft , I see couple of notebook tests are failing with this timeout, Do you know is there any issue with the pipelines?

**Comment by chlafreniere** [comment]
@ssreerama please check the "Azure Data Studio - Product Build" pipeline. If the same tests are failing there from last night's build, you can safely assume that this isn't related and check in.

**Comment by ssreerama** [comment]
Gotcha! these failing tests are found in last run of "Azure Data Studio - Product Build" pipeline!

**Pull request merged**